### PR TITLE
Project 2: Create and Remove books

### PR DESCRIPTION
**Pull request by crgc**
### Features

- Added the `redux-logger` middleware
- Configured the Redux Store and added two actions (**ADD** and **REMOVE**)
- Bound UI elements to React components, so that `actions` are dispatched when the user interacts with the app (either by adding a new book or removing an existing one)

### CODE REVIEW

File: src/components/AddBook.js [hunk 1]
```diff
@@ -1,29 +1,57 @@
...
-    </form>
-  </div>
-);
+import { v4 as uuidv4 } from 'uuid';
+import { useDispatch } from 'react-redux';
+import { addBook } from '../redux/books/books';
+
+const AddBook = () => {
+  const getElementById = (id) => document.getElementById(id);
+
```
Comment (oxenprogrammer) [inline src/components/AddBook.js]: **Changes Required**
- [x] The component should contain: - A local state with `title`, `author`. Every time a user changes the input value of either `title` or `author`, react should re-render in response to the state update. 
You can then pass the value of title and author to the function handling onSubmit to dispatch the action to the store.

File: src/pages/Books.js [hunk 2]
```diff
@@ -1,20 +1,9 @@
...
-      id: 2,
-      title: 'Among the Thugs',
-      author: 'Bill Buford',
-      category: 'Journalism',
-    },
-  ];
+  const books = useSelector((state) => state.booksReducer);
 
   return (
     <div className="book-section">
```
Comment (oxenprogrammer) [inline src/pages/Books.js:9]: **Changes Required**
- [x] This should be a `Book Component` that take these values as props. Let us refactor it into a Book Component and just import it here passing these values as props to it.

**Review by oxenprogrammer — CHANGES_REQUESTED** [review]
Hi @crgc , 

Good job so far!
I can see you're really putting effort to become a world-class software developer :+1:
There are some issues that you still need to address in order to move forward to the next project.
Trust me you got this! :muscle:


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Consider the optional changes nice to have features and only work on them when you have satisfied the required changes_

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

File: src/pages/Books.js (hunk 2 above)
Comment (crgc) [reply src/pages/Books.js:9]: Good observation.

File: src/components/AddBook.js (hunk 1 above)
Comment (crgc) [reply src/components/AddBook.js]: Done! Thank you - great review.

**Review by DeliceLydia — APPROVED** [review]
# Congratulations🎉 🎉 

## Hi @crgc , 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



### Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard).. If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**